### PR TITLE
ignore parameters with mismatched shape in transfer learning

### PR DESCRIPTION
**Pull request by jerrywgz**
ignore parameter with mismatched shape automatically.

### CODE REVIEW

File: ppdet/utils/checkpoint.py [hunk 1]
```diff
@@ -113,6 +116,17 @@ def load_params(exe, prog, path, ignore_params=[]):
     logger.info('Loading parameters from {}...'.format(path))
 
     ignore_list = None
+
+    if finetune_match:
+        all_var_shape = {}
+        for block in prog.blocks:
+            for param in block.all_parameters():
+                all_var_shape[param.name] = param.shape
+        state = _load_state(path)
```
Comment (willthefrog) [inline ppdet/utils/checkpoint.py]: please note this only works for single file check points, i.e., `some_model.pdparams`.

Comment (jerrywgz) [reply ppdet/utils/checkpoint.py]: after paddle 1.7, load_program_state supports the parameters from  [save_params, save_persistables, save_vars ]

**Comment by willthefrog** [comment]
ah, alright, I must be confusing it with some other issue

File: ppdet/utils/checkpoint.py (hunk 1 above)
Comment (willthefrog) [reply ppdet/utils/checkpoint.py]: ah, alright, I must be confusing it with some other issue

File: configs/yolov3_mobilenet_v1_fruit.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ weights: output/yolov3_mobilenet_v1_fruit/best_model
 num_classes: 3
 finetune_exclude_pretrained_params: ['yolo_output']
 use_fine_grained_loss: false
+# no need for `finetune_exclude_pretrained_params` if set `finetune_match` to true.
+finetune_match: true
```
Comment (qingqing01) [inline configs/yolov3_mobilenet_v1_fruit.yml]: finetune_match ->  finetune_exclude_mismatch_shape ?

File: ppdet/utils/checkpoint.py [hunk 3]
```diff
@@ -113,6 +116,17 @@ def load_params(exe, prog, path, ignore_params=[]):
...
+        state = {k: v for k, v in state.items() if v.shape == all_var_shape[k]}
+        fluid.io.set_program_state(prog, state)
+        return
```
Comment (qingqing01) [inline ppdet/utils/checkpoint.py]: 这段代码可以和下面代码合并，另 `ignore_params` 为shape不匹配的params

File: configs/yolov3_mobilenet_v1_fruit.yml [hunk 4]
```diff
@@ -11,6 +11,8 @@ weights: output/yolov3_mobilenet_v1_fruit/best_model
 num_classes: 3
 finetune_exclude_pretrained_params: ['yolo_output']
 use_fine_grained_loss: false
+# no need for `finetune_exclude_pretrained_params` if set `finetune_exclude_mismatch_shape` to true.
+finetune_exclude_mismatch_shape: true
```
Comment (qingqing01) [inline configs/yolov3_mobilenet_v1_fruit.yml]: 例子只需保留一个吧

File: docs/advanced_tutorials/TRANSFER_LEARNING_cn.md [hunk 5]
```diff
@@ -8,21 +8,23 @@
...
+2. 在 train.py的启动参数中设置`finetune_exclude_pretrained_params`或`finetune_exclude_mismatch_shape`。例如：
 
 ```python
 export PYTHONPATH=$PYTHONPATH:.
 export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
 python -u tools/train.py -c configs/faster_rcnn_r50_1x.yml \
                         -o pretrain_weights=https://paddlemodels.bj.bcebos.com/object_detection/faster_rcnn_r50_1x.tar \
-                           finetune_exclude_pretrained_params=['cls_score','bbox_pred']
+                           finetune_exclude_pretrained_params=['cls_score','bbox_pred'] \
+                           finetune_exclude_mismatch_shape=True
```
Comment (qingqing01) [inline docs/advanced_tutorials/TRANSFER_LEARNING_cn.md]: 例子分开写，给人的感觉是得设置2个参数。

File: configs/yolov3_mobilenet_v1_fruit.yml (hunk 2 above)
Comment (jerrywgz) [reply configs/yolov3_mobilenet_v1_fruit.yml]: deleted

File: ppdet/utils/checkpoint.py (hunk 3 above)
Comment (jerrywgz) [reply ppdet/utils/checkpoint.py]: Done

File: configs/yolov3_mobilenet_v1_fruit.yml (hunk 4 above)
Comment (jerrywgz) [reply configs/yolov3_mobilenet_v1_fruit.yml]: Done

File: docs/advanced_tutorials/TRANSFER_LEARNING_cn.md (hunk 5 above)
Comment (jerrywgz) [reply docs/advanced_tutorials/TRANSFER_LEARNING_cn.md]: Done

**Pull request merged**